static/js/components/ErrorMessage_test.js

```diff
@@ -0,0 +1,233 @@
+        });
+      });
+    });
+  });
+
+  describe('showing errors on pages', () => {
+    let renderComponent, helper, patchUserProfileStub, listenForActions;
+    let dashboardErrorActions;
+
+    const confirmErrorMessage = (div, codeMessage, extraInfo = '') => {
+      let alert = div.querySelector('.alert-message');
+      let messages = alert.getElementsByTagName('p');
+      assert.deepEqual(messages[0].textContent, codeMessage);
+      assert.deepEqual(messages[1].textContent, extraInfo);
+      assert.deepEqual(messages[2].textContent, contactExpectation);
+    };
+
+    beforeEach(() => {
+      helper = new IntegrationTestHelper();
+      renderComponent = helper.renderComponent.bind(helper);
+      patchUserProfileStub = helper.sandbox.stub(api, 'patchUserProfile');
+      listenForActions = helper.listenForActions.bind(helper);
+
+      dashboardErrorActions = [
+        RECEIVE_DASHBOARD_FAILURE,
+        RECEIVE_GET_USER_PROFILE_SUCCESS
+      ];
+
+      helper.profileGetStub.
+        withArgs(SETTINGS.username).
+        returns(
+          Promise.resolve(Object.assign({}, USER_PROFILE_RESPONSE, {
+            username: SETTINGS.username
+          }))
+        );
+    });
+
+    afterEach(() => {
+      helper.cleanup();
+    });
+
+    describe('dashboard page', () => {
+      it('error from the backend triggers error message in dashboard', () => {
+        helper.dashboardStub.returns(Promise.reject(ERROR_RESPONSE));
+
+        return renderComponent("/dashboard", dashboardErrorActions, false).then(([, div]) => {
+          confirmErrorMessage(div, `AB123 ${errorString}`, 'Additional info: custom error message for the user.');
+        });
+      });
+
+      it('the error from the backend does not need to be complete', () => {
+        let response = _.cloneDeep(ERROR_RESPONSE);
+        delete response.user_message;
+        helper.dashboardStub.returns(Promise.reject(response));
+
+        return renderComponent("/dashboard", dashboardErrorActions, false).then(([, div]) => {
+          confirmErrorMessage(div, `AB123 ${errorString}`);
+        });
+      });
+
+      it('the error from the backend does not need to exist at all as long as there is an http error', () => {
+        helper.dashboardStub.returns(Promise.reject({
+          errorStatusCode: 500
+        }));
+
+        return renderComponent("/dashboard", dashboardErrorActions, false).then(([, div]) => {
+          confirmErrorMessage(div, `500 ${errorString}`);
+        });
+      });
+
+      it('a regular response does not show the error', () => {
+        helper.dashboardStub.returns(Promise.resolve(DASHBOARD_RESPONSE));
+
+        return renderComponent("/dashboard").then(([, div]) => {
+          let message = div.getElementsByClassName('alert-message')[0];
+          assert.equal(message, undefined);
+        });
+      });
+    });
+
+    describe('profile page', () => {
+      it('shows an error when profile get has errored', () => {
```

Comment:
It looks like this test title is wrong, this is actually testing the spinner. Can you change the title? Not sure if it should be moved back or not, I'm fine either way